storage/tests/unit/test_blob.py

```diff
@@ -1612,23 +1618,25 @@ def _upload_from_file_helper(self, side_effect=None, **kwargs):
         num_retries = kwargs.get('num_retries')
         blob._do_upload.assert_called_once_with(
             client, stream, content_type,
-            len(data), num_retries, predefined_acl)
+            len(data), num_retries, predefined_acl,
+            extra_headers)
```

Comment:
`_do_upload` gets called with `extra_headers` as a keyword argument, so you need to have it as a keyword argument here, i.e.: `extra_headers=extra_headers`.